### ISSUE TITLE
ocilib::Date::operator inverted operator >= and <= 

### ISSUE THREAD

**Issue by ghost** [vrogier/ocilib#155]
Hello Vincent,

I have written a simple test case and found a "typo" in comparator for Date class. Library incorrectly evaluates operators >= and <= for date which in this example is in format "MM/DD/YYYY":
> 02/07/2018 > 02/01/2018: 1
> 02/07/2018 < 03/31/2018: 1
> 02/07/2018 >= 02/01/2018: 0
> 02/07/2018 <= 03/31/2018: 0

I have checked your code in ocilib_impl.hpp and found the error. This should be the correct definition of those operators.
```
inline bool Date::operator >= (const Date& other) const
{
    int res = Compare(other);

    return res == 0 || res > 0;
}

inline bool Date::operator <= (const Date& other) const
{
    int res = Compare(other);

    return res == 0 || res < 0;
}
```

**Comment by vrogier** [comment]
Fix committed in v4.5.2 branch

**Comment by brano543** [comment]
You are welcome. For the time being I have found a workaround by telling Oracle to output date in lexicographical format (YYYY-MM-DD) and compared two dates as string.
